src/main/java/uk/nhs/hee/tis/common/upload/service/AwsStorageService.java

```diff
@@ -157,18 +184,103 @@ public List<FileSummaryDto> listFiles(final StorageDto storageDto,
    * @throws AwsStorageException if there is a problem deleting the object
    */
   public void delete(final StorageDto storageDto) {
+    final ObjectMetadata objectMetadata = amazonS3
+        .getObjectMetadata(storageDto.getBucketName(), storageDto.getKey());
+    String metaDeleteType = objectMetadata == null
+        ? null : objectMetadata.getUserMetaDataOf(USER_METADATA_DELETE_TYPE);
+
+    if (metaDeleteType != null && metaDeleteType.equals(DeleteType.PARTIAL.name())) {
+      partialDelete(storageDto, objectMetadata);
```

Comment:
I think we might need a bit of explanation in the javadoc for this function to make it clear when partial- vs. hard-deletes are made.

Reply:
It's a good shout. Javadoc is updated with more details.